src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt

```diff
@@ -154,31 +156,40 @@ class SmartDocumentsTemplatesService @Inject constructor(
-                                    builder.equal(
-                                        root.get<ZaakafhandelParameters>(
-                                            SmartDocumentsTemplate::zaakafhandelParameters.name
-                                        )
-                                            .get<Long>("id"),
-                                        getZaakafhandelParametersId(zaakafhandelParametersUUID)
-                                    ),
-                                    builder.equal(
-                                        root.get<SmartDocumentsTemplateGroup>(
-                                            SmartDocumentsTemplate::templateGroup.name
+                    try {
+                        return entityManager.createQuery(
+                            query.select(root)
+                                .where(
+                                    builder.and(
+                                        builder.equal(
+                                            root.get<ZaakafhandelParameters>(
+                                                SmartDocumentsTemplate::zaakafhandelParameters.name
+                                            )
+                                                .get<Long>("id"),
+                                            getZaakafhandelParametersId(zaakafhandelParametersUUID)
+                                        ),
+                                        builder.equal(
+                                            root.get<SmartDocumentsTemplateGroup>(
+                                                SmartDocumentsTemplate::templateGroup.name
+                                            )
+                                                .get<String>(SmartDocumentsTemplate::smartDocumentsId.name),
+                                            templateGroupId
+                                        ),
+                                        builder.equal(
+                                            root.get<SmartDocumentsTemplate>(
+                                                SmartDocumentsTemplate::smartDocumentsId.name
+                                            ),
+                                            templateId
                                         )
-                                            .get<String>(SmartDocumentsTemplate::smartDocumentsId.name),
-                                        templateGroupId
-                                    ),
-                                    builder.equal(
-                                        root.get<SmartDocumentsTemplate>(SmartDocumentsTemplate::smartDocumentsId.name),
-                                        templateId
                                     )
                                 )
-                            )
-                    ).singleResult.informatieObjectTypeUUID
+                        ).singleResult.informatieObjectTypeUUID
+                    } catch (nre: NoResultException) {
```

Comment:
```suggestion
                    } catch (noResultsException: NoResultException) {
```

I don't like abbreviations when there is no clear benefit. ;-)